travis failure:

```
./components/tools/OmeroPy/test/integration/test_iquery.py:156:19: F821 undefined name 'CommentAnnotationI'
./components/tools/OmeroPy/test/integration/test_iquery.py:162:24: F812 list comprehension redefines 'x' from line 157
```

--- a/components/tools/OmeroPy/test/integration/test_iquery.py
+++ b/components/tools/OmeroPy/test/integration/test_iquery.py
@@ -167,7 +167,8 @@ class TestQuery(ITest):
         """
         params = ParametersI()
         params.addString("uuid", uuid)
-        rv = [x[0] for x in unwrap(self.query.projection(query_string, params))]
+        rv = self.query.projection(query_string, params)
+        rv = [x[0] for x in unwrap(rv)]
         assert len(rv) == 2
         assert "ome.model.annotations.CommentAnnotation" in rv
         assert "ome.model.annotations.TagAnnotation" in rv

--- a/components/tools/OmeroPy/test/integration/test_iquery.py
+++ b/components/tools/OmeroPy/test/integration/test_iquery.py
@@ -26,6 +26,7 @@
 
 from omero.testlib import ITest
 from omero.rtypes import unwrap, wrap
+from omero.model import CommentAnnotationI
 from omero.model import TagAnnotationI, ImageI, ImageAnnotationLinkI
 from omero.model import PermissionsI
 from omero.sys import ParametersI
@@ -153,9 +154,9 @@ class TestQuery(ITest):
 
     def testClassType(self):
         created = []
-        for x in (CommentAnnotationI, TagAnnotationI):
-            x = self.update.saveAndReturnObject(x())
-            created.append(x)
+        for Ann in (CommentAnnotationI, TagAnnotationI):
+            ann = self.update.saveAndReturnObject(Ann())
+            created.append(ann)
         query_string = """
         select type(a.class) from Annotation a
         """

--- a/components/tools/OmeroPy/test/integration/test_iquery.py
+++ b/components/tools/OmeroPy/test/integration/test_iquery.py
@@ -25,6 +25,7 @@
 """
 
 from omero.testlib import ITest
+from omero.rtypes import rstring
 from omero.rtypes import unwrap, wrap
 from omero.model import CommentAnnotationI
 from omero.model import TagAnnotationI, ImageI, ImageAnnotationLinkI
@@ -153,14 +154,20 @@ class TestQuery(ITest):
             assert result1[idx] == result2[idx]
 
     def testClassType(self):
+        uuid = self.uuid()
         created = []
         for Ann in (CommentAnnotationI, TagAnnotationI):
-            ann = self.update.saveAndReturnObject(Ann())
+            ann = Ann()
+            ann.setNs(rstring(uuid))
+            ann = self.update.saveAndReturnObject(ann)
             created.append(ann)
         query_string = """
         select type(a.class) from Annotation a
+        where a.ns = :uuid
         """
-        rv = [x[0] for x in unwrap(self.query.projection(query_string, None))]
+        params = ParametersI()
+        params.addString("uuid", uuid)
+        rv = [x[0] for x in unwrap(self.query.projection(query_string, params))]
         assert len(rv) == 2
         assert "ome.model.annotations.CommentAnnotation" in rv
         assert "ome.model.annotations.TagAnnotation" in rv

